src/main/kotlin/org/rust/ide/commenter/RsCommenter.kt

```diff
@@ -27,8 +41,100 @@ class RsCommenter : Commenter, CodeDocumentationAwareCommenter {
     override fun getBlockCommentPrefix(): String = "/*"
     override fun getBlockCommentSuffix(): String = "*/"
 
-    // for nested comments
-
+    // unused because we implement SelfManagingCommenter
     override fun getCommentedBlockCommentPrefix(): String = "*//*"
     override fun getCommentedBlockCommentSuffix(): String = "*//*"
+
+    override fun getBlockCommentPrefix(selectionStart: Int, document: Document, data: CommentHolder): String? =
+        blockCommentPrefix
+
+    override fun getBlockCommentSuffix(selectionEnd: Int, document: Document, data: CommentHolder): String? =
+        blockCommentSuffix
+
+    override fun getBlockCommentRange(
+        selectionStart: Int,
+        selectionEnd: Int,
+        document: Document,
+        data: CommentHolder
+    ): TextRange? =
+        SelfManagingCommenterUtil.getBlockCommentRange(selectionStart, selectionEnd, document, blockCommentPrefix, blockCommentSuffix)
+
+    override fun insertBlockComment(
+        startOffset: Int,
+        endOffset: Int,
+        document: Document,
+        data: CommentHolder?
+    ): TextRange {
+        val sequence = document.charsSequence
+        val start = CharArrayUtil.shiftForward(sequence, startOffset, " \t\n")
+        val end = CharArrayUtil.shiftBackward(sequence, endOffset - 1, " \t\n")
+        return SelfManagingCommenterUtil.insertBlockComment(start, end + 1, document, blockCommentPrefix, blockCommentSuffix)
+    }
+
+    override fun uncommentBlockComment(
+        startOffset: Int,
+        endOffset: Int,
+        document: Document,
+        data: CommentHolder?
+    ) =
+        SelfManagingCommenterUtil.uncommentBlockComment(startOffset, endOffset, document, blockCommentPrefix, blockCommentSuffix)
+
+    override fun isLineCommented(line: Int, offset: Int, document: Document, data: CommentHolder): Boolean {
+        return getStartLineComment(line, document, data.file)?.isEolComment ?: false
+    }
+
+    override fun commentLine(line: Int, offset: Int, document: Document, data: CommentHolder) {
+        document.insertString(offset, "// ")
```

Comment:
Should the code style be also respected when uncommenting? i.e. when `LINE_COMMENT_ADD_SPACE` is false, do not remove space after `//` even if there is a space?

Reply:
The default implementation checks the settings... and yes, I think we should. It may be reasonable for uncommenting an indented block:
```rust
//if foo {
//    bar(); // here the space can be wrongly removed
//}
```